client/app/app.js

```diff
@@ -472,66 +472,67 @@ angular
 
-                angular.forEach(templateUrlTab, (templateUrl) => {
-                    let routeTmp = templateUrl.substring(templateUrl.indexOf("/") + 1, templateUrl.lastIndexOf("/"));
-                    let index = routeTmp.lastIndexOf("/");
+                                if (_.isUndefined(value.noTranslations) && !value.noTranslations) {
+                                    if (value.templateUrl) {
+                                        templateUrlTab.push(value.templateUrl);
+                                    }
+                                    if (value.translations) {
+                                        translationsTab = _.union(translationsTab, value.translations);
+                                    }
+                                }
 
-                    while (index > 0) {
-                        translationsTab.push(routeTmp);
-                        routeTmp = routeTmp.substring(0, index);
-                        index = routeTmp.lastIndexOf("/");
-                    }
+                            });
+                        }
 
-                    translationsTab.push(routeTmp);
-                });
+                        angular.forEach(templateUrlTab, (templateUrl) => {
+                            let routeTmp = templateUrl.substring(templateUrl.indexOf("/") + 1, templateUrl.lastIndexOf("/"));
+                            let index = routeTmp.lastIndexOf("/");
 
-                // mmmhhh... It seems that we have to refresh after each time a part is added
+                            while (index > 0) {
+                                translationsTab.push(routeTmp);
+                                routeTmp = routeTmp.substring(0, index);
+                                index = routeTmp.lastIndexOf("/");
+                            }
 
-                translationsTab = _.uniq(translationsTab);
+                            translationsTab.push(routeTmp);
+                        });
 
-                state.resolve.translations = ["$translate", "$translatePartialLoader", function ($translate, $translatePartialLoader) {
-                    // load translation parts
-                    angular.forEach(translationsTab, (part) => {
-                        $translatePartialLoader.addPart(part);
-                    });
+                        // mmmhhh... It seems that we have to refresh after each time a part is added
 
-                    return $translate.refresh();
+                        translationsTab = _.uniq(translationsTab);
 
-                }];
+                        angular.forEach(translationsTab, (part) => {
+                            $translatePartialLoader.addPart(part);
+                        });
 
-                return translationsTab;
-            }
+                        return $translate.refresh();
+                    }
 
-            return null;
-        });
+                    return null;
+                }
+            }); // translation.addResolvable
```

Comment:
huh, this should not be needed, maybe you could extract some code to function so the `addResolvable` is shorter